js/bybit.js

```diff
@@ -3327,24 +3435,40 @@ module.exports = class bybit extends Exchange {
                 }
             } else if (api === 'private') {
                 this.checkRequiredCredentials ();
-                const timestamp = this.nonce ();
-                const query = this.extend (params, {
-                    'api_key': this.apiKey,
-                    'recv_window': this.options['recvWindow'],
-                    'timestamp': timestamp,
-                });
-                const sortedQuery = this.keysort (query);
-                const auth = this.rawencode (sortedQuery);
-                const signature = this.hmac (this.encode (auth), this.encode (this.secret));
-                if (method === 'POST') {
-                    body = this.json (this.extend (query, {
-                        'sign': signature,
-                    }));
+                const isOpenapi = url.indexOf ('openapi') >= 0;
+                const timestamp = this.milliseconds ();
+                if (isOpenapi) {
+                    let query = '';
+                    if (Object.keys (params).length) {
```

Comment:
This process is tricky, with your fix works on the PHP side but fails on Javascript (image attached)

From what I could test in PHP query needs to be empty '' and in JS needs to be an empty object {}.  That is the reason I was doing (Array.IsArray(...))

![image](https://user-images.githubusercontent.com/43336371/167368398-5891d015-af5a-4ee4-bb39-5e392ffc0d30.png)

Reply:
@kroitor Check my last commit, this is the only way I can make it work on the PHP side.

Reply:
I'll try to add a shorter solution to it.